### PR TITLE
Fix #2834: Remove js.GlobalScope

### LINKED ISSUE #2834: Remove js.GlobalScope

### PR DESCRIPTION
**Pull request by gzm0**
As a side effect, this:
- Removes binary compatibility for classes without loading spec.
- Makes loading spec checks strict (part of #1111).
- Adds a missing test.

### CODE REVIEW

**Comment by gzm0** [comment]
CI only for now.

**Comment by sjrd** [comment]
Looks good, but I believe it would be a good idea to rebase on top of #2831 once the latter has been merged.

**Comment by sjrd** [comment]
Well, it needs to be rebased, because conflicts ;)

**Comment by gzm0** [comment]
> Well, it needs to be rebased, because conflicts ;)

Was expecting that TBH :)

**Comment by gzm0** [comment]
Rebased.

File: compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala [hunk 1]
```diff
@@ -202,6 +202,32 @@ class JSInteropTest extends DirectTest with TestHelpers {
   }
 
   @Test
+  def noJSGlobalScopeAnnotOnNonJSNative: Unit = {
+
+    """
+    @ScalaJSDefined
```
Comment (sjrd) [inline compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala]: Oops. Rebase artifact :P

Comment (gzm0) [inline compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala]: Nice!

**Comment by gzm0** [comment]
Updated.

File: compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala [hunk 2]
```diff
@@ -202,6 +202,31 @@ class JSInteropTest extends DirectTest with TestHelpers {
...
+  def noJSGlobalScopeAnnotOnNonJSNative: Unit = {
+
+    """
+    @JSGlobalScope
+    object A extends js.Object
+    """ hasErrors
+    """
+      |newSource1.scala:6: error: Only native JS objects can have an @JSGlobalScope annotation.
+      |    @JSGlobalScope
+      |     ^
```
Comment (sjrd) [inline compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala]: One trivial incorrect line number in neg tests:
```
[error] Test org.scalajs.core.compiler.test.JSInteropTest.noJSGlobalScopeAnnotOnNonJSNative failed: should have right errors expected:<newSource1.scala:[6]: error: Only native...> but was:<newSource1.scala:[5]: error: Only native...>
[error]     at org.scalajs.core.compiler.test.util.TestHelpers$CompileTests.hasErrors(TestHelpers.scala:32)
[error]     at org.scalajs.core.compiler.test.JSInteropTest.noJSGlobalScopeAnnotOnNonJSNative(JSInteropTest.scala:210)
[error]     ...
```

**Comment by gzm0** [comment]
Updated (not tested locally due to time constraints).

**Pull request merged**